cmd/npmExecuteLint.go

```diff
@@ -0,0 +1,170 @@
+package cmd
+
+import (
+	"fmt"
+	"github.com/SAP/jenkins-library/pkg/command"
+	piperhttp "github.com/SAP/jenkins-library/pkg/http"
+	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/SAP/jenkins-library/pkg/npm"
+	"github.com/SAP/jenkins-library/pkg/piperutils"
+	"github.com/SAP/jenkins-library/pkg/telemetry"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+)
+
+type lintUtils interface {
+	Glob(pattern string) (matches []string, err error)
+
+	getExecRunner() execRunner
+	getGeneralPurposeConfig(configURL string)
+}
+
+type lintUtilsBundle struct {
+	*piperutils.Files
+	execRunner *command.Command
+	client     *piperhttp.Client
+}
+
+func newLintUtilsBundle() *lintUtilsBundle {
+	return &lintUtilsBundle{
+		Files:  &piperutils.Files{},
+		client: &piperhttp.Client{},
+	}
+}
+
+func (u *lintUtilsBundle) getExecRunner() execRunner {
+	if u.execRunner == nil {
+		u.execRunner = &command.Command{}
+		u.execRunner.Stdout(log.Writer())
+		u.execRunner.Stderr(log.Writer())
+	}
+	return u.execRunner
+}
+
+func (u *lintUtilsBundle) getGeneralPurposeConfig(configURL string) {
+	response, err := u.client.SendRequest(http.MethodGet, configURL, nil, nil, nil)
+	if err != nil {
+		log.Entry().Warnf("failed to download general purpose configuration: %v", err)
+		return
+	}
+
+	defer response.Body.Close()
+
+	content, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		log.Entry().Warnf("error reading %v: %w", response.Body, err)
```

Comment:
I am not sure about passing response.Body to Warnf(). Will this log the entire contents? Maybe that's the intention.

Reply:
Yeah at first I thought to provide more context but will probably just mess up the logs. I guess it's better to only log `err` then